lib/ansible/modules/user.py

```diff
@@ -3355,6 +3362,9 @@ def main():
                 if not os.path.isdir(parent):
                     path_needs_parents = True
 
+            if user.seuser:
+                user.check_seuser_support()
```

Comment:
I think it may be better to explicitly check if the system has SELinux and if it is enabled instead of relying on the platform type since it is possible for a system to support SELinux but it may be disabled or not installed.

```suggestion
                if user.seuser and not user.module.selinux_enabled():
                    user.module.warn(f"The 'seuser' parameter is not supported on {user.distribution or user.platform} "
                                     "as it lacks SELinux support and has been ignored."
```